core/server/src/main/java/alluxio/master/block/BlockMaster.java

```diff
@@ -77,21 +75,30 @@
 public final class BlockMaster extends AbstractMaster implements ContainerIdGenerable {
```

Comment:
Technically, I still don't think it is threadsafe, since the journal play back is not protected. The reason it is not protected is because the journal play back is always single threaded.